echoshader/new_version/echogram.py

```diff
@@ -183,56 +224,66 @@ def echogram_single_frequency(
             **kwargs: Additional keyword arguments.
 
         Returns:
-            holoviews.Image: The generated echogram visualization.
+            holoviews.DynamicMap: The generated echogram visualization with bounds.
         """
         if channel is not None:
             self.channel_select.value = channel
 
         if cmap is not None:
             self.color_map.value = cmap
 
-        list_camp = convert_string_to_list(self.color_map.value)
-        self.gram_opts["Image"]["cmap"] = (
-            list_camp if list_camp is not False else self.color_map.value
-        )
-
-        if vmin is not None and vmax is not None:
-            self.Sv_range_slider.start = vmin
-            self.Sv_range_slider.end = vmax
-            self.Sv_range_slider.value = (vmin, vmax)
+        self.gram_opts["Image"]["cmap"] = self.color_map.value
 
-        elif vmin is not None:
-            self.Sv_range_slider.start = vmin
-            old_vmax = self.Sv_range_slider.value[1]
-            self.Sv_range_slider.value = (vmin, old_vmax)
+        old_vmin = self.Sv_range_slider.value[0]
+        old_vmax = self.Sv_range_slider.value[1]
+        self.Sv_range_slider.value = (
+            vmin if vmin is not None else old_vmin,
+            vmax if vmax is not None else old_vmax,
+        )
+        self.gram_opts["Image"]["clim"] = self.Sv_range_slider.value
 
-        elif vmax is not None:
-            self.Sv_range_slider.end = vmax
-            old_vmin = self.Sv_range_slider.value[0]
-            self.Sv_range_slider.value = (old_vmin, old_vmax)
+        if link_to_track is not None:
+            self.link_mode_select.value = not link_to_track
+
+        if self.link_mode_select.value is False and self.positions_box is not None:
+            gram_plot = (
+                holoviews.Dataset(
+                    self.MVBS_ds.where(
+                        (self.MVBS_ds.longitude > self.positions_box.bounds[0])
+                        & (self.MVBS_ds.latitude > self.positions_box.bounds[1])
+                        & (self.MVBS_ds.longitude < self.positions_box.bounds[2])
+                        & (self.MVBS_ds.latitude < self.positions_box.bounds[3])
+                    ).sel(channel=self.channel_select.value)
+                )
+                .to(holoviews.Image, vdims=["Sv"], kdims=["ping_time", "echo_range"])
+                .opts(self.gram_opts)
+            )
 
-        self.gram_opts["Image"]["clim"] = self.Sv_range_slider.value
+        else:
+            gram_plot = (
```

Comment:
I think this is something I've commented on last time and you fixed? Please change all "gram" to "echogram" in your codebase, for readability
```suggestion
            echogram = (
```